packages/valory/skills/safe_deployment_abci/rounds.py

```diff
@@ -1,202 +0,0 @@
-    no_majority_event = Event.NO_MAJORITY
-    collection_key = get_name(SynchronizedData.participant_to_votes)
-    synchronized_data_class = SynchronizedData
-
-
-class FinishedSafeRound(DegenerateRound):
-    """A round that represents that the safe has been deployed"""
-
-
-class SafeDeploymentAbciApp(AbciApp[Event]):
-    """SafeDeploymentAbciApp
-
-    Initial round: RandomnessSafeRound
-
-    Initial states: {RandomnessSafeRound}
-
-    Transition states:
-        0. RandomnessSafeRound
-            - done: 1.
-            - round timeout: 0.
-            - no majority: 0.
-        1. SelectKeeperSafeRound
-            - done: 2.
-            - round timeout: 0.
-            - no majority: 0.
-        2. DeploySafeRound
-            - done: 3.
-            - deploy timeout: 1.
-            - failed: 1.
-        3. ValidateSafeRound
-            - done: 4.
-            - negative: 0.
-            - none: 0.
-            - validate timeout: 0.
-            - no majority: 0.
-        4. FinishedSafeRound
-
-    Final states: {FinishedSafeRound}
-
-    Timeouts:
-        round timeout: 30.0
-        validate timeout: 30.0
-        deploy timeout: 30.0
-    """
-
-    initial_round_cls: AppState = RandomnessSafeRound
-    transition_function: AbciAppTransitionFunction = {
-        RandomnessSafeRound: {
-            Event.DONE: SelectKeeperSafeRound,
-            Event.ROUND_TIMEOUT: RandomnessSafeRound,
-            Event.NO_MAJORITY: RandomnessSafeRound,
-        },
-        SelectKeeperSafeRound: {
-            Event.DONE: DeploySafeRound,
-            Event.ROUND_TIMEOUT: RandomnessSafeRound,
-            Event.NO_MAJORITY: RandomnessSafeRound,
-        },
-        DeploySafeRound: {
-            Event.DONE: ValidateSafeRound,
-            Event.DEPLOY_TIMEOUT: SelectKeeperSafeRound,
-            # NOTE: Consider the case where the keeper does send the tx but doesn't share the hash.
-            # We do not need to check for this! A simple round timeout will do here as the safe is
-            # either unusable by the keeper (requiring the other agent's to sign) or simply has signers which are
-            # different to the other agents and therefore they don't care about this safe instance.
-            Event.FAILED: SelectKeeperSafeRound,
-        },
-        ValidateSafeRound: {
-            Event.DONE: FinishedSafeRound,
-            Event.NEGATIVE: RandomnessSafeRound,
-            Event.NONE: RandomnessSafeRound,  # NOTE: unreachable
-            Event.VALIDATE_TIMEOUT: RandomnessSafeRound,
-            Event.NO_MAJORITY: RandomnessSafeRound,
-        },
-        FinishedSafeRound: {},
-    }
-    final_states: Set[AppState] = {FinishedSafeRound}
-    event_to_timeout: Dict[Event, float] = {
-        Event.ROUND_TIMEOUT: 30.0,
-        Event.VALIDATE_TIMEOUT: 30.0,
-        Event.DEPLOY_TIMEOUT: 30.0,
-    }
-    cross_period_persisted_keys: List[str] = [
-        get_name(SynchronizedData.safe_contract_address)
-    ]
```

Comment:
Now that this is removed here, do we need to add it somewhere else. I don't see any AbciApp that defines this as cross_period. Do we need to add it somewhere?

Reply:
Yes, that would be a good idea, for the reason that you mentioned in https://github.com/valory-xyz/price-oracle/pull/23#discussion_r1053645614. Maybe we should set some default cross-period keys, and the safe address would be one of them?

Reply:
Yes let’s do that.

Reply:
All default pre conditions

Reply:
Setting all default pre-conditions will not work, unfortunately, because they are defined by property name:

https://github.com/valory-xyz/open-autonomy/blob/10b215f6848c7bc25ca178038f4678cc57fca0db/packages/valory/skills/abstract_round_abci/base.py#L741-L746

Instead, the cross-period keys are implemented in such a way that they must be given by their key name in the db. This also means that there is a bug. In essence, if someone creates a property with a different name than the db key that the property is accessing, and then uses the `get_name()` to set it as a cross-period persisted key, then it will not work:

https://github.com/valory-xyz/open-autonomy/blob/aa9481cab797c2fe6ae5f6a0df4ea17b014a8a71/packages/valory/skills/reset_pause_abci/rounds.py#L58-L70

Reply:
Moreover, it does not make sense to have something called `cross_period_persisted_keys` in the db and let the `reset_pause_round` handle it. It might be better to handle it in the db's `create()` method.

Reply:
Done only for the safe address in 0466577aecd6596dfede187b4de9c02fa275be51. I believe that we should open an issue for the above comments.